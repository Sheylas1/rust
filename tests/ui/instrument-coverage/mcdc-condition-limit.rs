//@ edition: 2021
//@ min-llvm-version: 18
//@ revisions: good bad
//@ check-pass
//@ compile-flags: -Cinstrument-coverage -Zcoverage-options=mcdc -Zno-profiler-runtime

#[cfg(good)]
fn main() {
    let [a, b, c, d, e, f] = <[bool; 6]>::default();
    if a && b && c && d && e && f {
        core::hint::black_box("hello");
    }
}

#[cfg(bad)]
fn main() {
    let [a, b, c, d, e, f, g] = <[bool; 7]>::default();
    if a && b && c && d && e && f && g { //[bad]~ WARNING Conditions number of the decision
        core::hint::black_box("hello");
    }
}
