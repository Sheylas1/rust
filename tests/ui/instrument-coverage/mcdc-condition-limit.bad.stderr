warning: number of conditions in decision (7) exceeds limit (6)
  --> $DIR/mcdc-condition-limit.rs:18:8
   |
LL |     if a && b && c && d && e && f && g {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this decision will not be instrumented for MC/DC coverage

warning: 1 warning emitted

